src/clients/AcrossConfigStoreClient/AcrossConfigStoreClient.ts

```diff
@@ -193,36 +193,40 @@ export class AcrossConfigStoreClient extends BaseAbstractClient {
    * @param fromBlock Start block to search inclusive
    * @param toBlock End block to search inclusive. Defaults to MAX_SAFE_INTEGER, so grabs all disabled chain events
    * up until `latest`.
-   * @param allPossibleChains Returned list will be a subset of this list.
    * @returns List of chain IDs that have been enabled at least once in the block range. Sorted from lowest to highest.
    */
   getEnabledChainsInBlockRange(fromBlock: number, toBlock = Number.MAX_SAFE_INTEGER): number[] {
-    if (toBlock < fromBlock) {
+    // If our fromBlock is greater than our toBlock, then we have an invalid range.
+    if (fromBlock > toBlock) {
       throw new Error(`Invalid block range: fromBlock ${fromBlock} > toBlock ${toBlock}`);
     }
-    // Initiate list with all chains enabled at the fromBlock.
+
+    // Initiate list with all possible chains enabled at the toBlock while removing any chains
+    // that were disabled at the from block.
     const disabledChainsAtFromBlock = this.getDisabledChainsForBlock(fromBlock);
-    const enabledChainsAtFromBlock = this.getChainIdIndicesForBlock(fromBlock).filter(
+    const allPossibleChains = this.getChainIdIndicesForBlock(toBlock);
+    const enabledChainsInBlockRange = allPossibleChains.filter(
       (chainId) => !disabledChainsAtFromBlock.includes(chainId)
     );
 
-    // Update list of enabled chains with any of the candidate chains that have been removed from the
-    // disabled list during the block range.
-    return sortEventsAscending(this.cumulativeDisabledChainUpdates)
-      .reduce((enabledChains: number[], disabledChainUpdate) => {
-        if (disabledChainUpdate.blockNumber > toBlock || disabledChainUpdate.blockNumber < fromBlock) {
-          return enabledChains;
-        }
-        // If any of the possible chains are not listed in this disabled chain update and are not already in the
-        // enabled chain list, then add them to the list.
-        enabledChainsAtFromBlock.forEach((chainId) => {
-          if (!disabledChainUpdate.chainIds.includes(chainId) && !enabledChains.includes(chainId)) {
-            enabledChains.push(chainId);
+    // If there are any disabled chain updates in the block range, then we might need to update the list of enabled
+    // chains in the block range.
+    sortEventsAscending(this.cumulativeDisabledChainUpdates)
```

Comment:
Do we still need this sortEventsAscending..?

Reply:
It's actually not needed at all since our conditions are for something to exist at least once. Good call - I'll remove it

Reply:
Updated: https://github.com/across-protocol/sdk-v2/pull/350/commits/cb3626ed2043e624a0e7ae85e454fe498d55a19e